kubeflow/kubeflow
# Support K8s 1.22 in notebook controller

Fix https://github.com/kubeflow/kubeflow/issues/6366

Migrating to Kubebuilder v3 leads to the following changes:
- Add .dockerignore file.
- Upgrade Go version from v1.15 to v1.17.
- Adapt Makefile.
- Upgrade EnvTest to use K8s v1.22.
- Update PROJECT template.
- Migrate CRD apiVersion from v1beta to v1.
- Add livenessProbe and readinessProbe to controller manager.
- Upgrade controller-runtime from v0.2.0 to v0.11.0.

Other changes:
- Build image using public.ecr.aws registry instead of gcr.io.
- Update README.md documentation.
- Update 3rd party licences.
- Fix notebook.spec description.
- Add 3 sample notebooks (v1, v1alpha1 and v1beta1).

### Testing

1. Install the `notebooks.kubeflow.org` CRD:

```
make install
```

2. Deploy the notebook controller manager:

```
make deploy
```

3. Verify that the controller is running in the `notebook-controller-system` namespace:

```
$ kubectl get pods -l app=notebook-controller -n notebook-controller-system
NAME                                             READY   STATUS    RESTARTS   AGE
notebook-controller-deployment-564d76877-mqsm8   1/1     Running   0          16s
```

4. Create some notebooks (see config/samples).

Signed-off-by: Samuel Veloso <svelosol@redhat.com>
